tests/unit/s2n_handshake_test.c

```diff
@@ -434,6 +435,42 @@ int main(int argc, char **argv)
         }
     }
 
+    /* Ensure that a handshake can be performed after all file descriptors are closed */
+    {
+        /* A fork is created to ensure that closing file descriptors (like stdout) won't impact
+         * other tests.
+         */
+        pid_t pid = fork();
+        if (pid == 0) {
+            long max_file_descriptors = sysconf(_SC_OPEN_MAX);
+            EXPECT_TRUE(max_file_descriptors > -1);
+            int max_fd = MIN(INT_MAX, max_file_descriptors - 1);
+            for (int fd = 0; fd <= max_fd; fd++) {
+                close(fd);
+            }
```

Comment:
Sorry to keep harping on this loop, but since max_fd can be INT_MAX and your fd looping value is an int, what if max_fd is INT_MAX? Won't it wrap when you iterate and never actually end? See https://godbolt.org/z/j5qcoqaG8

This is why we generally just use size_t as the looping variable. You have to be very careful when you're writing a non standard loop.

Reply:
Yeah you're right, thanks!  I didn't think about what would happen with an iNT_MAX. I rewrote it to be a more normal loop to avoid this.